src/unix/macos.rs

```diff
@@ -0,0 +1,92 @@
+use crate::FileTime;
+use std::ffi::CString;
+use std::fs::{File, Metadata};
+use std::io;
+use std::os::unix::prelude::*;
+use std::path::Path;
+
+pub fn set_file_times(p: &Path, atime: FileTime, mtime: FileTime) -> io::Result<()> {
+    set_times(p, Some(atime), Some(mtime), false)
+}
+
+pub fn set_file_mtime(p: &Path, mtime: FileTime) -> io::Result<()> {
+    set_times(p, None, Some(mtime), false)
+}
+
+pub fn set_file_atime(p: &Path, atime: FileTime) -> io::Result<()> {
+    set_times(p, Some(atime), None, false)
+}
+
+pub fn set_file_handle_times(
+    f: &File,
+    atime: Option<FileTime>,
+    mtime: Option<FileTime>,
+) -> io::Result<()> {
+    let (atime, mtime) = match get_times(atime, mtime, || f.metadata())? {
```

Comment:
I think that we'll want to avoid `get_times` and use `UTIME_OMIT` to say that a particular time shouldn't be set

Reply:
BTW, it turns out that in darwin they've redefined `UTIME_OMIT` to `-2`. Go figure! [Here's the link](https://github.com/apple/darwin-xnu/blob/a449c6a3b8014d9406c2ddbdc81795da24aa7443/bsd/sys/stat.h#L541).